### DISCUSSION TITLE
celery run old code and new code at same time

### DISCUSSION THREAD

**Discussion by steinliber** [celery/celery#7535] [Q&A]
Hi,all!!
I use celery to run my django task. After i change my code yesterday, i find celery run both my old code and new code.So when i pass same parameter to task, sometimes i get error, sometimes it excutes successfully.  i have tryd purge, restart celery, but all uesless.
Can anyone tell me what happened

**Comment by DuncanBetts** [comment]
Purge should have purged everything (celery -A projectName purge), but if you're using the Django Database Scheduler in Django Admin area, and have not removed unwanted tasks using this, it might be re-instating them.
I'm not an expert, just trying to help.

**Comment by pranayberry** [comment]
Didi you find any solution to it? I am having same problem with my local machine and in my production server.

**Comment by dmitri-vlasov** [comment]
> Didi you find any solution to it? I am having same problem with my local machine and in my production server.

I faced with the same problem today. The steps that helped me:

1. Stop all workers.
2. Clear python cache in project dir. This command will clear __pychache__ content across all dirs:
`find . | grep -E "(__pycache__|\.pyc|\.pyo$)" | xargs rm -rf`
3. Restart a server.

**Comment by pkrakesh** [comment]
Same problem. The above solution is not working for me. Any help?

**Comment by max-vakario** [comment]
Would love to have a solution on that. That'a a big problem for me

**Comment by Vas12345** [comment]
Is there any fix for this issue? Or is there any workaround to stop making celery pick up old code.

**Comment by pkrakesh** [comment]
Make sure that you don't have any commented codes in any task. If there is any, delete it. This made my head spin and at last I found the problem was this.

**Reply by Abhishek-S24** [reply]
@pkrakesh was the issue resolved after that? For me when i pushed new changes in task the celery is exicuting old code. If this issue got resolved do let me know thank you.

**Comment by iwalucas** [comment]
So, I found that after killing (control+c) the process, it was still on the background. I had hundreds! so I killed it all and it started to get the new code again

**Comment by adarsh2911** [comment]
I have encountered with the same issue once, there may be chances that celery is running somewhere else too, please be sure your task is executing by the same celery worker.

**Comment by Abhishek-S24** [comment]
I am running  a email queue task using celery and rabbitmq as broker when i pushed new changes to the code celery is picking old code and failing the task when the task is triggered again then its taking new code. So is there anyway to resolve this , only new changes in task is getting affected and getting failed.
